group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Utils.java

```diff
@@ -235,4 +239,8 @@ public static ApiMessage messageOrNull(ApiMessageAndVersion apiMessageAndVersion
             return apiMessageAndVersion.message();
         }
     }
+
+    public static long hashSubscriptionMetadata(Map<String, TopicMetadata> subscriptionMetadata) {
+        return Murmur3.hash64(new TreeMap<>(subscriptionMetadata).toString().getBytes(StandardCharsets.UTF_8));
```

Comment:
This seems pretty inefficient. I wonder if we could compute the hash incrementally while we iterate on the (sorted) topic names. Then, for each topic we could add the topic id, the number of partitions and the (sorted) racks of each partitions. It would avoid building the TreeMap, avoid converting it to a String to finally transforms it into bytes. Have you considered it?